openvdb/tools/FastSweeping.h

```diff
@@ -0,0 +1,1432 @@
+///        off an iso-surface from an input SDF volume.
+///
+/// @return A shared pointer to the extension field defined on the active values in the
+///         input signed distance field.
+///
+/// @param sdfGrid  An approximate signed distance field to the specified iso-surface.
+///
+/// @param op       Functor with signature [](const Vec3R &xyz)->float that
+///                 defines the Dirichlet boundary condition, on the iso-surface,
+///                 of the field to be extended.
+///
+/// @param isoValue A value which defines a smooth iso-surface that
+///                 intersects active voxels in @a sdfGrid.
+///
+/// @param nIter    Number of iterations of the fast sweeping algorithm.
+///                 each performing 2^3 = 8 sweeps.
+///
+/// @note The only difference between this method and fogToEXT, defined above, is the
+///       convention of the sign of the signed distance field.
+///
+/// @details Topology of output grid is identical to that of the input grid, except
+///          active tiles in the input grid will be converted to active voxels
+///          in the output grid!
+///
+/// @throw  RuntimeError if the iso-surface does not intersected any active
+///         voxels or if it intersects any active tiles in @a sdfGrid.
+template<typename GridT, typename OpT>
+typename GridT::Ptr
+sdfToExt(const GridT &sdfGrid,
+         const OpT &op,
+         typename GridT::ValueType isoValue = 0,
+         int nIter = 1);
+
+/// @brief Computes the signed distance field and the extension of a scalar field,
+///        defined by the specified functor, off an iso-surface from an input FOG volume.
+///
+/// @return An array of two shared pointers to respectively the SDF and extension field
+///
+/// @param fogGrid  Scalar (floating-point) volume from which an
+///                 iso-surface can be defined.
+///
+/// @param op       Functor with signature [](const Vec3R &xyz)->float that
+///                 defines the Dirichlet boundary condition, on the iso-surface,
+///                 of the field to be extended.
+///
+/// @param isoValue A value which defines a smooth iso-surface that
+///                 intersects active voxels in @a fogGrid.
+///
+/// @param nIter    Number of iterations of the fast sweeping algorithm.
+///                 each performing 2^3 = 8 sweeps.
+///
+/// @note Strictly speaking a fog volume is normalized to the range [0,1] but this
+///       method accepts a scalar volume with an arbitary range, as long as the it
+///       includes the @a isoValue.
+///
+/// @details Topology of output grids are identical to that of the input grid, except
+///          active tiles in the input grid will be converted to active voxels
+///          in the output grids!
+///
+/// @throw  RuntimeError if the iso-surface does not intersected any active
+///         voxels or if it intersects any active tiles in @a fogGrid.
+template<typename GridT, typename OpT>
+std::array<typename GridT::Ptr, 2>
```

Comment:
Perhaps return a std::pair or GridPtrVec instead?

Reply:
sure, why do you prefer that over a fixed-length array?

Reply:
Well the prior, in my mind, implies a direct relationship between the two elements. The latter simply because it's already a VDB typedef. This is minor though